apps/nrm/behaviours/get-saved-report.js

```diff
@@ -58,11 +58,58 @@ const getReports = async(req) => {
   }
 };
 
+/**
+ * Get local page data
+ *
+ * We need to feedback to the user the outcome of the read
+ * request. So we check the sessionModel variable 'report-read-success'
+ * to determine whether the request was successful.
+ *
+ * @const reportsRequestStatus - if true or false then user is sent to page and will see
+ * content related to a sucessful / failed read from their list of reports.
+ *
+ * Otherwise an object declaring that no records were found will be sent
+ *
+ * @param {object} req - request object
+ *
+ * @returns {object} - containing the outcome report
+ * save process
+ */
+const getLocalPageData = (req) => {
+  let reportsRequestStatus = {
+    reportSaved: false,
+  };
+
+  if (req.sessionModel.get('report-read-success')) {
+    reportsRequestStatus.reportSaved = true;
+  }
+
+  if (req.sessionModel.get('no-report-found')) {
+    return {
+      reportsNotFound: true,
```

Comment:
Returning various types of incompatible objects feels weird.
Would it be better to throw an error and bubble it up?